zeeguu/api/api/nlp.py

```diff
@@ -22,3 +23,37 @@ def do_some_spacy():
     tokens = SpacyWrappers[language].tokenize_sentence(phrase)
 
     return json_result(tokens)
+
+
+# ---------------------------------------------------------------------------
+@api.route("/create_confusion_words", methods=("POST",))
+# ---------------------------------------------------------------------------
+@cross_domain
+@with_session
+def create_confusion_words():
+    original_sent = request.form.get("original_sent", "")
+    language = request.form.get("language")
+
+    if language not in SpacyWrappers.keys():
+        return "Language not supported"
+    
+    noise_words = NoiseWordsGenerator[language].generate_confusion_words(original_sent)[0]
```

Comment:
[0] is to only get the confusion word list, this is the return of the function:

`return list(confusion_set), pos_pick, id_to_add`

The reason I returned the pos_pick is that we wanted to keep track of what POS is being used in the exercises for future improvements, like selecting POS based on each student's needs. The id_to_add, is the id of the word that was picked, so if we wanted to store the word associated with it. 

I am OK with making it only return the list, at this point that's the main thing we are looking at. Otherwise, I could make it a dictionary, so it's clear what is being retrieved?

Reply:
Dictionary is great!